sympy/solvers/tests/test_solvers.py

```diff
@@ -164,6 +164,13 @@ def test_solve_args():
     assert solve([(x + y)**2 - 4, x + y - 2]) == [{x: -y + 2}]
     # - linear
     assert solve((x + y - 2, 2*x + 2*y - 4)) == {x: -y + 2}
+    # When one or more args are Boolean
+    assert solve([True, Eq(x, 0)], [x], dict=True) == [{x: 0}]
+    assert solve([Eq(x, x), Eq(x, 0), Eq(x, x+1)], [x], dict=True) == []
+    assert solve([Eq(x, x+1), x < 2], x) == False
```

Comment:
Can you use `is False` or `is S.false` here?

Other things can compare equal to False:
```julia
In [4]: 0 == False                                                                                                                                            
Out[4]: True
```